cdap-docs/reference-manual/source/release-notes.rst

```diff
@@ -30,6 +30,545 @@ Cask Data Application Platform Release Notes
    :backlinks: none
    :depth: 2
 
+`Release 4.1.0 <http://docs.cask.co/cdap/4.1.0/index.html>`__
+=============================================================
+
+New Features
+------------
+
+Revamped Log System
+...................
+
+- :cask-issue:`CDAP-7676` - Added a prototype implementation for a rolling HDFS log
+  appender.
+
+- :cask-issue:`CDAP-8108` - Revised the CDAP Log Appender to use `Logback
+  <http://logback.qos.ch/>`__\ 's Appender interface.
+
+- :cask-issue:`CDAP-8231` - The log file cleaner thread will remove metadata and, for
+  successfully deleted metadata entries, it will delete the corresponding log files. The log
+  file cleaner thread will only remove the metadata entries for the old (pre-4.1.0) log
+  format.
+
+- :cask-issue:`CDAP-8261` - Logs collected by the CDAP Log Appender will be stored at a
+  common ``<cdap>/logs`` path, owned by the cdap user. For security, it is readable only by
+  the cdap user.
+
+- :cask-issue:`CDAP-8428` - Added additional metrics about the status of the log
+  framework.
+
+Replication, Resiliency, and Zero-downtime
+..........................................
+
+- :cask-issue:`CDAP-8025` - Added a tool to check a cluster's replication status.
+
+- :cask-issue:`CDAP-8032` - CDAP context methods will now be retried according to a
+  program's retry policy.
+
+- :cask-issue:`CDAP-8037` - Added a ``master.manage.hbase.coprocessors`` setting that can be
+  set to false on clusters where the CDAP coprocessors are deployed on every HBase node.
+
+Dataset Improvements
+....................
+- :cask-issue:`CDAP-7596` - Added the ability to reuse an existing file system location
+  and Hive table when creating a partitioned file set.
+
+- :cask-issue:`CDAP-7597` - Added configuring the CDAP Explore database and table name for
+  a dataset using dataset properties.
+
+- :cask-issue:`CDAP-7683` - Added a tool that pre-builds and loads the HBase coprocessors
+  required by CDAP onto HDFS.
+
+- :cask-issue:`CDAP-7685` - Provided SPI hooks that users can implement for performing
+  HBase DDL operations.
+
+- :cask-issue:`CDAP-8070` - Added control of group ownership and permissions through
+  dataset properties.
+
+Secure Impersonation and Authorization
+......................................
+- :cask-issue:`CDAP-8110` - Added support for fine-grained impersonation at the CDAP
+  application, dataset, and stream level.
+
+- :cask-issue:`CDAP-8355` - Impersonated namespaces can be configured to disallow the
+  impersonation of the namespace owner when running CDAP Explore queries.
+
+Brand-new CDAP UI
+.................
+
+- :cask-issue:`CDAP-8075` - The CDAP UI added one-step deploy wizards for the Cask Market.
```

Comment:
Should we rephrase this and the next few from with the user as a subject? 

"Users can now deploy applications and plugins from the Cask Market with a single click, instead of downloading them from the  market and then uploading them."